### PR TITLE
Add User CRUD with Api Model

### PR DESCRIPTION
**Pull request by Sergyo97**

### CODE REVIEW

File: src/main/java/leafout/backend/apimodel/UserRequest.java [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+import leafout.backend.model.Feedback;
+import lombok.*;
+
+@Builder
+@Getter
+@Setter
+public class UserRequest {
+
+    private String name;
```
Comment (Alejoguzm07) [inline src/main/java/leafout/backend/apimodel/UserRequest.java:20]: Javadoc

File: src/main/java/leafout/backend/apimodel/UserResponse.java [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+import leafout.backend.model.Ticket;
+import lombok.*;
+
+import java.util.List;
+
+@Builder
+@Getter
+public class UserResponse {
+
+    private String name;
```
Comment (Alejoguzm07) [inline src/main/java/leafout/backend/apimodel/UserResponse.java:22]: Javadoc

File: src/main/java/leafout/backend/controller/UserController.java [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
+
+@RestController
+@RequestMapping(value = "/users")
```
Comment (Alejoguzm07) [inline src/main/java/leafout/backend/controller/UserController.java:19]: Javadoc

File: src/main/java/leafout/backend/controller/UserController.java [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+public class UserController {
+
+    @Autowired
+    private UserServiceImpl userService;
+
+    @GetMapping
+    public ResponseEntity<?> getAll(){
```
Comment (Alejoguzm07) [inline src/main/java/leafout/backend/controller/UserController.java:26]: Explicit return type

File: src/main/java/leafout/backend/controller/UserController.java [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+        try {
+            List<User> users =  userService.getAll();
```
Comment (Alejoguzm07) [inline src/main/java/leafout/backend/controller/UserController.java]: final modifier

File: src/main/java/leafout/backend/controller/UserController.java [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+            return new ResponseEntity<>(users, HttpStatus.ACCEPTED);
+        }catch (Exception e){
```
Comment (Alejoguzm07) [inline src/main/java/leafout/backend/controller/UserController.java]: Explicit exception type

File: src/main/java/leafout/backend/service/IUserService.java [hunk 7]
```diff
@@ -0,0 +1,16 @@
...
+
+import leafout.backend.model.User;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
+
+public interface IUserService {
+
+    List<User> getAll();
```
Comment (Alejoguzm07) [inline src/main/java/leafout/backend/service/IUserService.java]: Javadoc to be implemented by the impl

**Review by Alejoguzm07 — CHANGES_REQUESTED** [review]

**Review by Alejoguzm07 — APPROVED** [review]

**Review by QSARJP — APPROVED** [review]

**Pull request merged**